Add support for passing event params for filtering events

rickerbh opened the pull request:
This PR adds support for passing parameters to the `get-stored-events` function, to perform server-side filtering.

If there is anything else you'd like me to add to get this PR merged just let me know.

ukriish commented:
@rickerbh Hey, this looks fine. I just realised there isn't a test for this, can you add one? I'll be happy to merge this PR then.

rickerbh commented:
@krish8664 I've extracted the functionality to generate the body for the event query and added a test for this. 

Sadly, it's different enough from the mail body generation that it _might_ warrant its own function. I could look at refactoring the `gen-body` function to account for multipart, form-params, and query-params if you like, but I'd need to add another parameter to it so the default param type can be passed through.
